src/sniffer.c

```diff
@@ -596,8 +600,12 @@ static int HashUpdate(HsHashes* hash, const byte* input, int sz)
         ret = wc_ShaUpdate(&hash->hashSha, input, sz);
 #endif
 #ifndef NO_MD5
-    if (ret == 0)
-        wc_Md5Update(&hash->hashMd5, input, sz);
+    if (ret == 0) {
+        ret = wc_Md5Update(&hash->hashMd5, input, sz);
+        if (ret !=0) {
+            return ret;
```

Comment:
This is not the pattern in the rest of the function, to return early.  Follow existing patterns.